.gitignore

```diff
@@ -1,4 +1,9 @@
+.DS_Store
+
 **/target
 
 # Don't add the generated MDBook artifacts
 docs/book/
+
+# Any rust programs in examples/
+examples/simple-non-wasm/programs/counter-rs/Cargo.lock
```

Comment:
We should add

```
out
```

to this `.gitignore`.

in order to disclude the `out` directory generated by `forc build` which just contains the compiler output artifacts. This should help to stop the `examples/simple-non-wasm/programs/counter/out/` from being included, as well as any future forc project output artifacts.

Reply:
@mitchmindtree  
- I'll defer to you on whether or not that's a forc convention (I can go either way).
- I only kept the out/ because it includes the abi-json (which is needed by the example)
  - I could definitely always copy the abi-json out of that folder, but wanted the user to see the full structure of the example